inspirehep/modules/warnings/ext.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Extension that suppresses warnings.
+
+By default Invenio-Base enables deprecation warnings. This extension
+disables them, unless ``DEBUG_SHOW_DEPRECATION_WARNINGS`` is ``True``.
+
+This extension also disables SQLAlchemy warnings, unless the variable
+``DEBUG_SHOW_SQLALCHEMY_WARNINGS` is set to ``True``.
```

Comment:
No breaking change in the Invenio API was ever anticipated by a warning, yet breaking API changes have happened all the time, and will continue to happen because they don't care about us. You just don't see them because they're caught by nightly and fixed on the spot before the next day.

The list of warnings that are safe to ignore is exactly this PR.